common/src/fetcher/mod.rs

```diff
@@ -134,6 +162,14 @@ impl Fetcher {
 
         log::debug!("Response Status: {}", response.status());
 
+        // Check for rate limiting
+        if let Some(retry_after) =
+            get_retry_after_from_response_header(&response, self.default_retry_after)
+        {
+            log::warn!("Rate limited (429), retry after: {:?}", retry_after);
```

Comment:
I wouldn't warn here. Info is fine. As nothing will go wrong, we handle this gracefully here. We might warn about this in the fetcher, where we handle this.